src/FSharp.Stats/HTest.fs

```diff
@@ -0,0 +1,77 @@
+﻿namespace FSharp.Stats.Testing
+
+
+module HTest =
+
+    open FSharp.Stats
+    // H-test / one-way ANOVA of ranks 
+    // input : seq{seq<float>} 
+    let htest (samples : seq<#seq<float>>) = 
+        // calculating n for each group 
+        let n = Seq.map Seq.length samples |> Seq.map float  
+    
+        // preparing samples for ranking and calculating samplesize 
+        let allValues = 
+            samples 
+            |> Seq.concat
+            |> Seq.toArray
+    
+        let samplesize = Seq.length allValues |> float
+    
+        // ranking all values 
+        let ranked = FSharp.Stats.Rank.rankAverage allValues
+    
+        let valuesAndRanks = Array.zip allValues ranked 
+    
+        // match ranks with each group 
+        let rankingOfEachGroup data = 
+            data 
+            |> Seq.choose (fun x ->   
+                Array.tryFind (fun (i,j) -> i = x) valuesAndRanks
+                )
+            |> Seq.map snd
+            |> Seq.sum 
+            |> float 
+        let ranks = Seq.map rankingOfEachGroup samples 
+    
+        // counting ties in data 
+    
+        let ties = 
+            valuesAndRanks
+            |> Seq.countBy id 
+            |> Seq.filter (fun (i,j) -> j > 1 )
+            |> Seq.map snd
+            |> Seq.map float 
+    
+        if ties = seq [] then 
```

Comment:
I think this if branch decision is incorrect. If there are no ties, no corretion factor must be calculated. I fixed the issue in commit 6f2cf997f48da6ab1ee594e0ddf3856f77395ca2.